mkosi/__init__.py

```diff
@@ -1545,9 +1546,49 @@ def build_initrd(context: Context) -> Path:
     return config.output_dir / config.output
 
 
+def identify_cpu(root: Path) -> tuple[Optional[Path], Optional[Path]]:
+    for entry in Path("/proc/cpuinfo").read_text().split("\n\n"):
+        vendor_id = family = model = stepping = None
+        for line in entry.splitlines():
+            key, _, value = line.partition(":")
+            key = key.strip()
+            value = value.strip()
+
+            if key == "vendor_id":
+                vendor_id = value
+            elif key == "cpu family":
+                family = int(value)
+            elif key == "model":
+                model = int(value)
+            elif key == "stepping":
+                stepping = int(value)
+
+        if vendor_id is not None and family is not None and model is not None and stepping is not None:
```

Comment:
Do we need `is not None` here or is just checking for truthy good enough?

Reply:
In most cases that should be alright, since I don't think the values can actually be empty (though I added a guard for that now) and the numbers will probably not be zero, but I think having an explicit dummy type here makes this more robust.